File: Municipal stage 2017-2018/4. Ah, the road.pas
//
// Решение на 100 баллов
//

program PR;
var x0, y0, r, Xa, Ya, Xb, Yb, l, d , AB, AO, BO, s: real;
    lk: string;
    f,g: text;

begin
  assing(f, 'input.txt')
  reset(f);
  readln(f, x0, y0, r);
  readln(f, Xa, Ya, Xb, Yb);
  close(f);
  
  AB:= sqrt(sqr(Ya-Yb)+sqr(Xa-Xb));
  AO:= sqrt(sqr(Ya-YO)+sqr(Xa-XO));
  BO:= sqrt(sqr(YO-Yb)+sqr(XO-Xb));
  
  s:= (sqr(AO)-sqr(BO)+sqr(AB))/(2*AB);
  d:= sqrt(sqr(AO)-sqr(s));
  l:= 2*sqrt(sqr(r)-sqr(d));
  
  assign(g, 'output.txt');
  rewrite(g);
  writeln(g, l:0:3);
  close(g);
end.

File: Municipal stage 2017-2018/4. Ah, the road.py
#
#  Решение на 100 баллов
#


inp = open('input.txt', 'r')
def input(): return inp.readline()
out = open('output.txt', 'w')
def print(s): return out.write(str(s))


x0, y0, r = map(float, input().split())
x1, y1, x2, y2 = map(float, input().split())

AB = ((y1 - y2) ** 2 + (x1 - x2) ** 2) ** 0.5
AO = ((y1 - y0) ** 2 + (x1 - x0) ** 2) ** 0.5
BO = ((y0 - y2) ** 2 + (x0 - x2) ** 2) ** 0.5

s = (AO ** 2 - BO ** 2 + AB ** 2) / (2 * AB)
d = (AO ** 2 - s ** 2) ** 0.5
l = 2 * ((r ** 2 - d ** 2) ** 0.5)

print(round(l, 3))
